### PR TITLE
Allow Node.js LTS ‘Hydrogen’ (v18) release version

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR allows Node.js LTS ‘Hydrogen’ (v18) to install, run and build **GOV.UK Frontend**

Changes included:

1. **Updating [.browserslistrc](https://github.com/alphagov/govuk-frontend/blob/main/.browserslistrc#L16)** for Node.js 18
So Babel is aware we need to support only `node 18`

2. **Updating [.nvmrc](https://github.com/alphagov/govuk-frontend/blob/main/.nvmrc)** for Node.js 18
So tools like [nvm](https://github.com/nvm-sh/nvm#readme) and [asdf](https://github.com/asdf-vm/asdf#readme) allow the `lts/hydrogen` LTS release

3. **Bump the [package.json `engines.node` range](https://github.com/alphagov/govuk-frontend/blob/main/package.json#L6)**
Ensures `npm install` and other scripts check the local Node.js version is allowed

3. **Widen the [package.json `engines.npm` range](https://github.com/alphagov/govuk-frontend/blob/main/package.json#L7)**
So we're ready for `npm@9` as it's being [backported to Node.js 18 in December](https://github.com/npm/statusboard/issues/587)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "name": "govuk-frontend-repository",
   "description": "Used only for the development of GOV.UK Frontend, see `package/package.json` for the published `package.json`",
   "engines": {
-    "node": "^16.13.0",
-    "npm": "^8.1.0"
+    "node": "^16.13.0 || ^18.12.0",
+    "npm": "^8.1.0 || ^9.1.0"
```
Comment (colinrotherham) [inline package.json:7]: I've included `npm@9` as it's being [backported to Node.js 18 in December](https://github.com/npm/statusboard/issues/587)

More info via [**npm CLI v9 is now generally available**](https://github.blog/changelog/2022-11-09-npm-cli-v9-is-now-generally-available/) from GitHub Changelog

**Comment by colinrotherham** [comment]
We should merge [`node-sass@8`](https://github.com/sass/node-sass/releases/tag/v8.0.0) first to avoid slow install times:

* https://github.com/alphagov/govuk-frontend/pull/3015

The [`node-sass@8`](https://github.com/sass/node-sass/releases/tag/v8.0.0) release has binaries for Node.js 16 and 18
The [`node-sass@7`](https://github.com/sass/node-sass/releases/tag/v7.0.0) release has no binaries for Node.js 18

Updating avoids a slow [`node-gyp`](https://github.com/nodejs/node-gyp#readme) compile step during `npm ci`

**Comment by 36degrees** [comment]
As discussed in dev catch up, I think we can switch to only supporting v18 as the current LTS and stop worrying about v16 entirely.

This matches what we did for [v16](https://github.com/alphagov/govuk-frontend/pull/2574), [v14](https://github.com/alphagov/govuk-frontend/pull/2249), [v12](https://github.com/alphagov/govuk-frontend/pull/1660) and [v10](https://github.com/alphagov/govuk-frontend/pull/1177).

**Comment by colinrotherham** [comment]
> As discussed in dev catch up, I think we can switch to only supporting v18 as the current LTS and stop worrying about v16 entirely.
> 
> This matches what we did for [v16](https://github.com/alphagov/govuk-frontend/pull/2574), [v14](https://github.com/alphagov/govuk-frontend/pull/2249), [v12](https://github.com/alphagov/govuk-frontend/pull/1660) and [v10](https://github.com/alphagov/govuk-frontend/pull/1177).

@36degrees Sounds good. How's it looking now?

**Comment by 36degrees** [comment]
Testing locally I was seeing issues running on my M1 Macbook Pro:

```
$ npm start     

> start
> gulp dev

Error: Node Sass does not yet support your current environment: OS X Unsupported architecture (arm64) with Node.js 18.x
For more information on which environments are supported please see:
https://github.com/sass/node-sass/releases/tag/v8.0.0
    at module.exports (/Users/oliver.byford/Code/govuk-frontend/node_modules/node-sass/lib/binding.js:13:13)
    at Object.<anonymous> (/Users/oliver.byford/Code/govuk-frontend/node_modules/node-sass/lib/index.js:13:35)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/Users/oliver.byford/Code/govuk-frontend/tasks/gulp/compile-assets.js:7:35)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
```

I've tried following the [instructions in the running locally doc](https://github.com/alphagov/govuk-frontend/blob/95b6b70e8188d71e47e896b2acb68d9709929f1f/docs/contributing/running-locally.md#fixing-nodejs-18-errors) but they make no difference.

However, running `npm rebuild node-sass` seems to have fixed it. Worth updating the docs?

Also worth flagging in the dev channel once this has merged.

**Comment by colinrotherham** [comment]
@36degrees Ha, you're a mind reader 😊 

I had it committed last week, but was worried it wouldn't pass code review so dropped it! I thought the "long way" (full `node_module` delete and reinstall) was preferred for a technical reason.

You might have seen variations with the `--force` flag logged in the past too? Context: https://github.com/sass/node-sass/commit/57c8b590fef43a6ac815b56ddce91ceb318c9818

Pushed up a new version for review 👍

**Comment by colinrotherham** [comment]
@36degrees Do you mind checking the docs update on https://github.com/alphagov/govuk-frontend/pull/3018/commits/2a851f8d0f0584f5a7a168d453290d08c4fe9c8b?

I'll put an update on Slack regarding this one too:

* https://github.com/alphagov/govuk-design-system/pull/2430

☝️ Although that PR won't throw errors (only warnings) [like we do](https://github.com/alphagov/govuk-frontend/blob/main/.npmrc#L1)

**Pull request merged**